core/src/main/scala/chisel3/internal/package.scala

```diff
@@ -131,6 +131,21 @@ package object internal {
     case leaf => leaf.probeInfo.nonEmpty
   }
 
+  private[chisel3] def requireCompatibleDestinationProbeColor(
+    dest:         Data,
+    errorMessage: String = ""
```

Comment:
```suggestion
    errorMessage: => String = ""
```
Making this by name will save the logic to construct the String at the call site (at the cost of constructing one object).

Reply:
Clever! Updated in https://github.com/chipsalliance/chisel/pull/3744/commits/1bc988918d02a3648358a349d52a350e9778f37a.